### PR TITLE
[ENGG-5058] fix: typescript errors in ApiClient/utils

### PR DESCRIPTION
**Pull request by nafees87n**
## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed potential runtime errors when responses lack content-type information by ensuring content type is treated consistently.
  * Improved test-generation reliability for API and GraphQL responses with missing or undefined headers.
  * Made response processing more robust across the API client to better handle edge cases.

* **Chores**
  * Aligned helper behavior to reflect possible missing header values, improving runtime consistency.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by nafees87n** [comment]
@coderabbitai review

File: app/src/features/apiClient/screens/apiClient/utils.ts [hunk 1]
```diff
@@ -767,7 +773,10 @@ export const processRecordsForDuplication = (
     const record = queue.shift()!;
 
     if (record.type === RQAPI.RecordType.COLLECTION) {
-      const newId = apiClientRecordsRepository.generateCollectionId(`(Copy) ${record.name}`, record.collectionId);
+      const newId = apiClientRecordsRepository.generateCollectionId(
+        `(Copy) ${record.name}`,
+        record.collectionId ?? undefined
+      );
```
Comment (Aarushsr12) [inline app/src/features/apiClient/screens/apiClient/utils.ts:779]: keeping undefined as fallback Id won't break in firebase ?

Comment (nafees87n) [reply app/src/features/apiClient/screens/apiClient/utils.ts:779]: The underlying function doesn't even use it. Its there to satisfy the interface

Comment (Aarushsr12) [reply app/src/features/apiClient/screens/apiClient/utils.ts:779]: understood
